engine/runtime-compiler/src/main/scala/org/enso/compiler/pass/analyse/alias/Graph.scala

```diff
@@ -343,13 +343,13 @@ object Graph {
   /** A representation of a local scope in Enso.
     *
     * @param childScopes all scopes that are _direct_ children of `this`
-    * @param occurrences all symbol occurrences in `this` scope
+    * @param occurrences all symbol occurrences in `this` scope indexed by the identifier of the name
     * @param allDefinitions all definitions in this scope, including synthetic ones.
     *                       Note that there may not be a link for all these definitions.
     */
   sealed class Scope(
     var childScopes: List[Scope]             = List(),
-    var occurrences: Set[Occurrence]         = Set(),
+    var occurrences: Map[Id, Occurrence]     = HashMap(),
```

Comment:
How effective is `HashMap[Int, Occurence]`? Is it really using boxed `java.lang.Integer` as keys or is there an optimized version for `int` keys? Found [IntMap](https://www.scala-lang.org/api/2.13.3/scala/collection/immutable/IntMap.html) which claims to be _superseded by HashMap_.

Reply:
Yes, it's boxing. Locally benchmarking `IntMap` version appears to be within the margin of error. So, since it really appears to be _superseded_ by `HashMap` I would leave it as is.

Reply:
OK, boxing may cost us more memory, but that is not our primary concern right now.